### PR TITLE
MM-15253 Refactor of create-issue modal.

### PR DESCRIPTION
**Pull request by crspeller**
A refactor of the create-issue modal as part of the larger changes related to the subscribe modal. 
Adds ability to render arbitrary fields but functionality is disabled at the moment to keep consistency with current implementation.

### CODE REVIEW

File: webapp/src/components/jira_field.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        obeyRequired: true,
+    };
+
+    handleChange = (id, value) => {
+        if (this.props.onChange) {
+            this.props.onChange(id, value);
+        }
+    };
+
+    allowedValueToSelectValue = (allowedValue) => {
```
Comment (levb) [inline webapp/src/components/jira_field.jsx]: Maybe a comment? Took me a couple minutes to figure out what this function did. Also, maybe slightly different name, `makeSelectValueFromAllowedValue`, or maybe even just `makeSelectValue = (allowedValue)...` would be more intention-revealing?

File: webapp/src/components/modals/create_issue/create_issue.jsx [hunk 2]
```diff
@@ -30,10 +32,11 @@ export default class CreateIssueModal extends PureComponent {
     static propTypes = {
         close: PropTypes.func.isRequired,
         create: PropTypes.func.isRequired,
-        getMetadata: PropTypes.func.isRequired,
         post: PropTypes.object,
         theme: PropTypes.object.isRequired,
         visible: PropTypes.bool.isRequired,
+        jiraMetadata: PropTypes.object,
```
Comment (levb) [inline webapp/src/components/modals/create_issue/create_issue.jsx]: maybe, jiraCreateIssueMetadata?

File: webapp/src/components/modals/create_issue/create_issue.jsx [hunk 3]
```diff
@@ -30,10 +32,11 @@ export default class CreateIssueModal extends PureComponent {
...
+        getMetadata: PropTypes.func.isRequired,
```
Comment (levb) [inline webapp/src/components/modals/create_issue/create_issue.jsx]: ditto

File: webapp/src/action_types/index.js [hunk 4]
```diff
@@ -1,8 +1,12 @@
...
-export {
-    CreateTypes,
-};
\ No newline at end of file
+export default {
+    CLOSE_CREATE_MODAL: `${PluginId}_close_create_modal`,
+    OPEN_CREATE_MODAL: `${PluginId}_open_create_modal`,
+    RECEIVED_CONNECTED: `${PluginId}_connected`,
+
+    RECEIVED_METADATA: `${PluginId}_received_metadata`,
```
[the remark this answers is not in the document]
Comment (crspeller) [reply webapp/src/action_types/index.js]: Ya. I can rename this stuff. The CREATE ones I didn't name just moved :P

File: webapp/src/components/jira_field.jsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import PropTypes from 'prop-types';
+
+import ReactSelectSetting from 'components/react_select_setting';
+import Input from 'components/input';
+
+export default class JiraField extends React.PureComponent {
+    static propTypes = {
+        id: PropTypes.object.isRequired,
+        field: PropTypes.object.isRequired,
+        obeyRequired: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (crspeller) [reply webapp/src/components/jira_field.jsx]: No it's for the subscribe modal.

**Pull request merged**